### PR TITLE
Add touch option to hide cursor on any touch input

### PR DESCRIPTION
**Pull request by frink**
I found this fork with additions to your project from a link in a random Stack Overflow post. It was 6 months old with no PR yet, so I thought it would be good to create one and get the code merged in. Seems like a useful addition...

### CODE REVIEW

**Comment by Airblader** [comment]
Thanks, but content from SO is to be assumed to be licensed CC BY-SA 3.0, I cannot simply merge this in, sadly. You'd have to ping the author on SO. Do you have a link to it?

**Comment by frink** [comment]
I think you misunderstood...

I found a reference link on StackOverflow that pointed to this fork on GitHub while trolling for answers on a touchscreen problem. I didn't write the code or copy it. I merely noted that a PR should not be done and the code deserved to be pushed back upstream...

To be clear, the PR is from a fork of your own repository hosted on GitHub with the same MIT license and even the same copyright notice. Everything is license MIT - Copyright Ingo Bürk. I've edited my comments above to make it clear that no code is license CC-BY-SA 3.0. From a license perspective, I do not believe there is any problem in merging this code.

SEE: https://github.com/nowrep/unclutter-xfixes/blob/master/LICENSE

I reviewed the code and it seemed reasonable so I created a pull request to help both of you and @nowrep connect and get a more useful project rather than divergent forking hell that can result from smaller projects on GitHub. Was just trying to help you harvest some changes back upstream.

Sorry for the confusion.
Cheers!

**Comment by Airblader** [comment]
Ah, OK, sorry for the misunderstanding then. I will reopen for now and have to look into this later as I'm quite busy right now. Thanks for considering to upstream such changes, though, it's much appreciated!

**Comment by frink** [comment]
It looks pretty simple. I don't see any real conflicts in the source. It's just git getting confused I think...

**Comment by Airblader** [comment]
Given that the remote HEAD is behind, I think git is right here and this needs to be rebased.

File: src/unclutter.c [hunk 1]
```diff
@@ -93,6 +94,7 @@ static void parse_args(int argc, char *argv[]) {
         { "jitter", required_argument, 0, 0 },
         { "exclude-root", no_argument, 0, 0 },
         { "ignore-scrolling", no_argument, 0, 0 },
+        { "touch", no_argument, 0, 0 },
```
Comment (Airblader) [inline src/unclutter.c]: "touch" somewhat suggests that it simply includes touch events, but this isn't the case; rather, it does the opposite of what we do for other input events. So I think it'd be better if we could name this `--hide-on-touch` (and everywhere else accordingly)

File: src/unclutter.c [hunk 2]
```diff
@@ -185,7 +190,7 @@ static void parse_args(int argc, char *argv[]) {
 }
 
 static void print_usage(char *argv[]) {
-    fprintf(stderr, "Usage: %s [--timeout <n>] [--jitter <radius>] [--exclude-root] [--ignore-scrolling] [-b|--fork] [-v|--version] [-h|--help]", argv[0]);
+    fprintf(stderr, "Usage: %s [--timeout <n>] [--jitter <radius>] [--exclude-root] [--ignore-scrolling] [--touch] [-b|--fork] [-v|--version] [-h|--help]", argv[0]);
```
Comment (Airblader) [inline src/unclutter.c]: Please also update the man page to document this option properly.

**Comment by Airblader** [comment]
Yeah, you'll have to make a fork for that. :-)

**Comment by frink** [comment]
Ok will do this week.

**Comment by chocolatemelt** [comment]
Is this still happening? The touch option is super handy, but it doesn't seem to be merged upstream yet. I don't mind making these fixes if Frink hasn't already.

**Comment by Airblader** [comment]
From my side everything's ready, the PR just has to be updated so it can be merged.

**Comment by nowrep** [comment]
Opened new pull request #52

**Pull request closed without merging**